src/tokenizedSubaccounts/PPTSA.sol

```diff
@@ -0,0 +1,559 @@
+    if (cashBalance <= 0) {
+      revert PPT_MustHavePositiveCash();
+    }
+    uint basePrice = _getBasePrice();
+
+    _verifyFee(tradeData.maxFee, basePrice);
+
+    if (tradeData.trades[0].price > basePrice.multiplyDecimal($.ppParams.worstSpotBuyPrice)) {
+      revert PPT_SpotLimitPriceTooHigh();
+    }
+
+    int cost = tradeData.trades[0].price.toInt256().multiplyDecimal(tradeData.trades[0].amount);
+    int bufferedBalance = cashBalance.multiplyDecimal($.ppParams.spotTransactionLeniency);
+    if (cost > bufferedBalance) {
+      revert PPT_BuyingTooMuchCollateral();
+    }
+  }
+
+  function _verifyCollateralSell(IRfqModule.RfqOrder memory tradeData, BaseTSAAddresses memory tsaAddresses)
+    internal
+    view
+  {
+    PPTSAStorage storage $ = _getPPTSAStorage();
+
+    int cashBalance = tsaAddresses.subAccounts.getBalance(subAccount(), tsaAddresses.cash, 0);
+    if (cashBalance >= 0) {
+      revert PPT_MustHaveNegativeCash();
+    }
+
+    uint basePrice = _getBasePrice();
+
+    _verifyFee(tradeData.maxFee, basePrice);
+
+    if (tradeData.trades[0].price < basePrice.multiplyDecimal($.ppParams.worstSpotSellPrice)) {
+      revert PPT_SpotLimitPriceTooLow();
+    }
+
+    // cost is positive, balance is negative
+    uint cost = tradeData.trades[0].price.multiplyDecimal(tradeData.trades[0].amount.toUint256());
+    int bufferedBalance = cashBalance.multiplyDecimal($.ppParams.spotTransactionLeniency);
+
+    // We make sure we're not selling more $ value of collateral than we have in debt
+    if (cost > bufferedBalance.abs()) {
+      revert PPT_SellingTooMuchCollateral();
+    }
+  }
+
+  function _verifyRfqExecute(IRfqModule.RfqOrder memory tradeData) internal view {
+    PPTSAStorage storage $ = _getPPTSAStorage();
+    if (tradeData.trades.length != 2) {
+      revert PPT_InvalidParams();
+    }
+    if (tradeData.trades[0].asset != tradeData.trades[1].asset) {
+      revert PPT_InvalidParams();
+    }
+
+    (, uint strike1,) = OptionEncoding.fromSubId(tradeData.trades[0].subId.toUint96());
+    (, uint strike2,) = OptionEncoding.fromSubId(tradeData.trades[1].subId.toUint96());
+    if (strike1 < strike2 || (strike1 - strike2).toInt256() != _getPPTSAStorage().ppParams.strikeDiff) {
+      revert PPT_StrikePriceOutsideOfDiff();
+    }
+
+    // TODO: if the amount is less than 0 does that indicate a sell?
+    if (tradeData.trades[0].amount != -tradeData.trades[1].amount || tradeData.trades[0].amount < 0) {
+      revert PPT_InvalidTradeAmountForMaker();
+    }
+    // trade array is from MM side so MM is selling call spreads.
+    // so we should expect the first trade to be a sell on the lower strike price,
+    // and the second trade to be a buy on the higher strike price
+    if (strike1 >= strike2) {
+      revert PPT_InvalidParams();
+    }
+    // base balance is sUSDe
+    (int amountUsedInTrades, uint baseBalance, int cashBalance) = _getSubAccountStats();
```

Comment:
What is needed for the checks here:

1) value of currently open options
2) MtM of subaccount
3) value of options being opened now (`totalCostOfTrade` you have below)

You could probably shortcut the value calculation by doing: `MtM - (baseBalance * basePrice)` to save on recomputing option prices multiple times (it's already done in the MtM calc)

Reply:
Hmmm are you saying that the validation for `baseBalance.multiplyDecimal($.ppParams.maxBuyPctOfTVL)`  isn't correct? I'm not too sure where the MtM of a subaccount comes into play and how that comes into play with the `baseBalance`. We'll probably clear this up during our meeting today. If not I'll put more details on my question.